### DISCUSSION TITLE
Unable to obtain access token via GitHub Enterprise

### DISCUSSION THREAD

**Discussion by zjpiazza** [brettcannon/gidgethub#155] [Q&A] [answered]
Hello,

I am trying to use gidgethub to build a GitHub App for some automation tasks. I was able to install and test the app on public GitHub without any problems. However, when I tried to port the application over to our company's GH Enterprise instance, I'm getting an exception thrown on the **get_installation_access_token** method. The response just says: **`gidgethub.BadRequest: Not Acceptable`**

If I use the **get_jwt** function myself and then use Postman or equivalent tool, I am able to POST a request to the access_tokens endpoint and see a token returned to me. Here is the relevant snippet:

```
@routes.post('/')
async def main(request):
    try:
        body = await request.read()
        event = sansio.Event.from_http(request.headers, body, secret=GH_SECRET)
        print('GH delivery ID', event.delivery_id, file=sys.stderr)

        if event.event == "ping":
            return web.Response(status=200)
        async with aiohttp.ClientSession() as session:
            gh = gh_aiohttp.GitHubAPI(session, "zachary-piazza", base_url='https://github.hpe.com/api/v3')
            access_token = await get_installation_access_token(
                gh, installation_id=GH_INSTALLATION_ID, app_id=GH_APP_ID,
                private_key=GH_PRIVATE_KEY
            )
            gh = gh_aiohttp.GitHubAPI(session, "zachary-piazza",
                                      oauth_token=access_token['token'],
                                      cache=cache, base_url='https://github.hpe.com/api/v3')
            # Give GitHub some time to reach internal consistency.
            await asyncio.sleep(1)
            await router.dispatch(event, gh)
        try:
            print('GH requests remaining:', gh.rate_limit.remaining)
        except AttributeError:
            pass
        return web.Response(status=200)
    except Exception as exc:
        traceback.print_exc(file=sys.stderr)
        return web.Response(status=500)
```

Am I doing something wrong in my my call to **get_installation_access_token**? Any help here would be appreciated. Thank you!

**Comment by brettcannon** [comment]
@Mariatta do you have any ideas?

**Reply by zjpiazza** [reply]
My apologies for not posting to the correct location

**Reply by brettcannon** [reply]
@zjpiazza absolutely no problem! We are still just testing out Discussions.

**Reply by Mariatta** [reply]
@zjpiazza, I'm assuming that you are using GitHub enterprise server v 2.22 or v 3?

It looks like, the `Create installation access token` is no longer a "preview". When it was implemented initially , I had to pass in the `application/vnd.github.machine-man-preview+json` accept header, but looking at more recent documentation at [GitHub.com](https://docs.github.com/en/rest/reference/apps#create-an-installation-access-token-for-an-app) and the documentation for [GitHub Enterprise server v 3] (https://docs.github.com/en/enterprise-server@3.0/rest/reference/apps), this endpoint is no longer under preview, and that we should pass in the `application/vnd.github.v3+json` accept header instead.

https://github.com/brettcannon/gidgethub/blob/4774a896f87023bcd3954cdf072656e69ca3e9ea/gidgethub/apps.py#L34

If I look at the documentation for [GitHub enterprise server v 2.21](https://docs.github.com/en/enterprise-server@2.21/rest/reference/apps#create-an-installation-access-token-for-an-app), it still says to use the preview header. 

To fix this, I would need to adjust the code such that it passes the `application/vnd.github.v3+json` accept header correctly.

**Reply by Mariatta** [reply]
So I think this is an actual issue, but I don't know to convert it back into an issue.
Anyway, I will work on a PR for it.

**Reply by Mariatta** [reply]
PR created at #156

**Comment by Mariatta** [comment, marked as answer]
The fix has been released to PyPI https://pypi.org/project/gidgethub/5.0.1/
On GitHub: https://github.com/brettcannon/gidgethub/releases/tag/5.0.1
Changelog https://gidgethub.readthedocs.io/en/latest/changelog.html
